### PR TITLE
Add early stopping, Update Prototype MNIST defaults, Add epoch value calculations to Prototype

### LINKED ISSUE #39: Phase 2 of training ProtoPNet reduces classification accuracy to chance
And phase 3 training doesn't raise the classification rate back to where it was before.

My hypothesis is that there is some problem with how I implemented the prototype push in phase 2.

Screencaps coming eventually for visual evidence of this issue (and maybe a tensorboard log file)

### PR DESCRIPTION
**Pull request by DrKwint**
Fixes issue #49, fixes issue #39 

Description:
- Added early stopping object to common and implemented it in prototype
- Changed prototype MNIST defaults to fix #39 (turns out dimension was too high)
- Added epoch value calculations for definitive test of performance

### CODE REVIEW

File: pyroclast/prototype/prototype.py [hunk 1]
```diff
@@ -118,33 +138,63 @@ def run_minibatch(epoch, batch, phase, is_train=True):
...
+        accuracy_numerator = tf.reduce_sum(
+            tf.cast(tf.equal(prediction, labels), tf.int32))
+        denominator = x.shape[0]
+        return loss_numerator, accuracy_numerator, denominator
 
     ### PHASE 1
     # run training loop
-    if debug: print("PHASE 1 - TRAINING CONV STACK AND PROTOTYPES")
-    for epoch in range(epochs_phase_1):
+    print("PHASE 1 - TRAINING CONV STACK AND PROTOTYPES")
```
Comment (ihowell) [inline pyroclast/prototype/prototype.py]: We might want to consider starting to use Python loggers: https://docs.python.org/3.7/library/logging.html
